server/controllers/list.js

```diff
@@ -190,15 +218,8 @@ const getListData = (req, resp) => {
       }
     })
     .then(() => {
-      const {
-        _id,
-        adminIds,
-        cohortId,
-        description,
-        isArchived,
-        items,
-        name
-      } = list;
+      const { _id, adminIds, cohortId, description, isArchived, name } = list;
+      const items = getItemsForList(userId, list);
```

Comment:
Why do you get list items alredy here? If the list `isArchived` this data won't be needed.